transport/handler.go

```diff
@@ -20,7 +20,39 @@
 
 package transport
 
-import "context"
+import (
+	"context"
+	"fmt"
+)
+
+// Type is an enum of RPC types
+type Type int
+
+const (
+	// Unary type
+	Unary Type = iota + 1
+	// Oneway type
+	Oneway
+)
+
+func (t Type) String() string {
+	switch t {
+	case Unary:
+		return "Unary"
+	case Oneway:
+		return "Oneway"
+	default:
+		return fmt.Sprintf("Type(%v)", int(t))
+	}
+}
+
+// HandlerSpec holds a handler and its Type
+type HandlerSpec struct {
+	Type Type
+
+	UnaryHandler  UnaryHandler
+	OnewayHandler OnewayHandler
+}
```

Comment:
Actually, instead of exposing the fields, consider making these private and
exposing accessors and constructors.

```
type HandlerSpec struct { .. }

func (HandlerSpec) Type() Type { .. }
func (HandlerSpec) Unary() UnaryHandler { .. }
func (HandlerSpec) Oneway() OnewayHandler { .. }

func NewUnaryHandlerSpec(UnaryHandler) HandlerSpec { ... }
func NewOnewayHandlerSpec(OnewayHandler) HandlerSpec { .. }
```

Besides making usage easier elsewhere, it'll also ensure we don't construct
invalid HandlerSpec objects.